File: WPServer/WPProjectServer/WPProjectServer/DatabaseObjects/Condition.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WPProjectServer.DatabaseObjects
{
    public class Condition : DatabaseObject
    {
        Conditions condition { get; set; }

        public override string ToString()
        {
            return condition.ToString();
        }

        public override bool Equals(object obj)
        {
            if(obj is Condition)
                return condition.Equals(((Condition)obj).condition);
            else if(obj is Conditions)
                return condition.Equals((Conditions)obj);
            return false;
        }
    }

    public enum Conditions
    {
        Avaliable = 1,

        Missing = 2,

        Damaged = 3,

        Destroyed = 4
    }
}

File: WPWeb/Library/src/main/java/models/Condition.java
package models;

/**
 * Created by Szymon on 2015-10-13.
 */
public class Condition extends DatabaseObject
{
    Conditions condition;

    public String ToString()
    {
        return String.valueOf(condition);
    }

    public boolean equals(Condition cond)
    {
        return condition.equals(cond.condition);
    }


    public boolean equals(Conditions conds)
    {
        return condition.equals(conds);
    }
}